src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -114,124 +131,183 @@ private CompilationDiff(ImmutableArray<Diagnostic> diagnostics, string originalP
+            static BuildDataFiles createBuildArtifacts(string outputPath, string assemblyFileName, BuildInfo buildInfo)
+            {
+                var assemblyName = Path.GetFileNameWithoutExtension(assemblyFileName);
+                var assemblyFilePath = Path.Combine(outputPath, assemblyFileName);
+                var buildDataFiles = new BuildDataFiles(
+                    AssemblyMdvFilePath: Path.Combine(outputPath, assemblyName + ".mdv"),
+                    PdbMdvFilePath: Path.Combine(outputPath, assemblyName + ".pdb.mdv"),
+                    ILFilePath: Path.Combine(outputPath, assemblyName + ".il"),
+                    PdbXmlFilePath: Path.Combine(outputPath, assemblyName + ".pdb.xml"),
+                    CustomDataFilePath: Path.Combine(outputPath, "custom-data.txt"));
+
+                Directory.CreateDirectory(outputPath);
+                File.WriteAllBytes(assemblyFilePath, buildInfo.AssemblyBytes);
+
+                // This is deliberately named .extracted.pdb instead of .pdb. A number of tools will look
+                // for a PDB with the name assemblyName.pdb. Want to make explicitly sure that does not 
+                // happen and such tools always correctly fall back to the embedded PDB. 
+                var pdbFilePath = Path.Combine(outputPath, assemblyName + ".extracted.pdb");
+                writeAllBytes(pdbFilePath, new Span<byte>(buildInfo.PdbMetadataReader.MetadataPointer, buildInfo.PdbMetadataReader.MetadataLength));
+
+                createMetadataVisualization(buildDataFiles.AssemblyMdvFilePath, buildInfo.AssemblyMetadataReader);
+                createMetadataVisualization(buildDataFiles.PdbMdvFilePath, buildInfo.PdbMetadataReader);
+                createDataFile(buildDataFiles.CustomDataFilePath, buildInfo.AssemblyReader, buildInfo.PdbMetadataReader);
+
+                var pdbToXmlOptions = PdbToXmlOptions.ResolveTokens
+                    | PdbToXmlOptions.ThrowOnError
+                    | PdbToXmlOptions.ExcludeScopes
+                    | PdbToXmlOptions.IncludeSourceServerInformation
+                    | PdbToXmlOptions.IncludeEmbeddedSources
+                    | PdbToXmlOptions.IncludeTokens
+                    | PdbToXmlOptions.IncludeMethodSpans;
+
+                using var pdbXmlStream = File.Create(buildDataFiles.PdbXmlFilePath);
+                PdbToXmlConverter.ToXml(
+                    new StreamWriter(pdbXmlStream),
+                    pdbStream: new UnmanagedMemoryStream(buildInfo.PdbMetadataReader.MetadataPointer, buildInfo.PdbMetadataReader.MetadataLength),
+                    peStream: new MemoryStream(buildInfo.AssemblyBytes),
+                    options: pdbToXmlOptions,
+                    methodName: null);
+
+                Process.Start(@"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\ildasm.exe", $@"{assemblyFilePath} /all /out={buildDataFiles.ILFilePath}").WaitForExit();
```

Comment:
randomly wondering: would `ProcessStartInfo.CreateNoWindow` let us prevent ildasm from popping up and stealing focus when this tool is running in the background.